### DISCUSSION TITLE
Turn on heater in SHT31-D?

### DISCUSSION THREAD

**Discussion by redm123** [arendst/Tasmota#15407] [Q&A] [answered]
Hi all
I bought a SHT31-D (Adafruit) temperature and humidity sensor. It's working fine reporting both. It has a heater to evaporate condensation, which I would like to use. Is there a way to turn that heater on and off?

The code of the driver module (https://github.com/arendst/Tasmota/blob/development/tasmota/xsns_14_sht3x.ino, if that is the right one...) does not suggest it's possible. But maybe I'm looking in the wrong place, or there is some other way to do it?

**Comment by sfromis** [comment, marked as answer]
You're looking the right place, and you've already answered the question :-) Ready to submit a PR?

With Berry on ESP32, and scripter, you do have some I2C features, but I do not know if it would be unproblematic to inject extra commands. Of course heater and temperature sensor would imply that reading temperature values would be unworkable in some period....

**Reply by redm123** [reply]
> Ready to submit a PR?

Hehe.. ;) I wouldn't say "no". Not right at the moment. But once I have the hardware wired up and installed, I can/will certainly take a look. 
As I have no idea about Tasmota development, do you have any pointers what approach I would use for this? I imagine some command I would execute by some rule, based on some condition (e.g. turn on if humidity >80%). Something based on a `Sensor*` command, like there exist for other sensors (https://tasmota.github.io/docs/Commands/#sensors)? Although they more look like configuration commands...

> Of course heater and temperature sensor would imply that reading temperature values would be unworkable in some period....

Yea, the impact and effect of the heater would be subject for experimentation. But according to the sensor spec there is also a status command. So one could maybe query this and supply the heater state with the regular tele updates... Then consumers can decide whether to use the temperature value or ignore it.
